crates/model/src/interaction.rs

```diff
@@ -1,10 +1,37 @@
+use ethcontract::Bytes;
 use primitive_types::{H160, U256};
 use serde::{Deserialize, Serialize};
 
+pub type EncodedInteraction = (
+    H160,           // target
+    U256,           // value
+    Bytes<Vec<u8>>, // callData
+);
+
+pub trait Interaction: std::fmt::Debug + Send + Sync {
```

Comment:
This trait lived in the solver before. But now, we wanna share it between all components